input/gen_messages.py

```diff
@@ -0,0 +1,205 @@
+#!/usr/bin/env python3
+
+# Copyright © 2022 Red Hat, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Generates messages to be consumed by Insights Results Aggregator.
+
+This script read input message (that should be correct or incorrect, according
+to needs) and generates bunch of new messages derived from input one. Each
+generated message can be updated if needed - Org ID can changed, cluster ID can
+changed as well etc.
+
+Types of possible input message modification:
+    * Org ID (if enabled by CLI flag -g)
+    * Account number (if enabled by CLI flag -a)
+    * Cluster ID (if enabled by CLI flag -c)
+
+It is also possible to specify pattern for output message filenames. For example:
+generated_message_{}.json
+
+
+usage: gen_messages.py [-h] [-i INPUT] [-o OUTPUT] [-r REPEAT] [-g] [-a] [-c] [-v]
+
+optional arguments:
+  -h, --help            show this help message and exit
+  -i INPUT, --input INPUT
+                        Specification of input file
+  -o OUTPUT, -output OUTPUT
+                        Specification of pattern of output file names
+  -r REPEAT, --repeat REPEAT
+                        Number of generated files
+  -g, --org-id          Enable organization ID modification
+  -a, --account-number  Enable account number modification
+  -c, --cluster-id      Enable cluster ID modification
+  -v, --verbose         Make messages verbose
+"""
+
+# Link to generated documentation for this script:
+# <https://redhatinsights.github.io/insights-results-aggregator-utils/packages/gen_messages.html>
+
+import json
+import sys
+import random
+import uuid
+
+from argparse import ArgumentParser
+
+
+# ranges for attributes that can be modified
+MIN_ORG_ID = 0
+MAX_ORG_ID = 100000000
+
+MIN_ACCOUNT_NUMBER = 0
+MAX_ACCOUNT_NUMBER = 100000000
+
+
+def cli_arguments():
+    """Retrieve all CLI arguments provided by user."""
+
+    # First of all, we need to specify all command line flags that are
+    # recognized by this tool.
+    parser = ArgumentParser()
+
+    # All supported command line arguments and flags
+    parser.add_argument(
+        "-i",
+        "--input",
+        dest="input",
+        default="input.json",
+        required=False,
+        help="Specification of input file",
+    )
+
+    parser.add_argument(
+        "-o",
+        "-output",
+        dest="output",
+        default="output{}.json",
```

Comment:
Just a nitpick. Maybe we should use `output_{}.json` instead? That way the base filename and index will be more separated.

Reply:
ok

Reply:
fixed